server/modules/monitor/mysql_mon.c

```diff
@@ -1126,28 +1128,41 @@ static void set_master_heartbeat(MYSQL_MONITOR *handle, MONITOR_SERVERS *databas
         return;
     }
 
-    /* create the maxscale_schema database */
-    if (mysql_query(database->con, "CREATE DATABASE IF NOT EXISTS maxscale_schema"))
-    {
-        MXS_ERROR("[mysql_mon]: Error creating maxscale_schema database in Master server"
-                  ": %s", mysql_error(database->con));
-
+    /* check if the maxscale_schema database and replication_heartbeat table exist */
+    if (mysql_query(database->con, "SELECT table_name FROM information_schema.tables WHERE table_schema = 'maxscale_schema' AND table_name = 'replication_heartbeat'")) {
```

Comment:
Please ensure that no line is longer than 110 chars. Manually wrap if necessary.